pecos/xmc/xlinear/model.py

```diff
@@ -437,37 +438,56 @@ def predict(
             kwargs:
                 beam_size (int, optional): override the beam size specified in the model.
                     Default None to disable overriding
                 only_topk (int, optional): override the only topk specified in the model
                     Default None to disable overriding
                 post_processor (str, optional):  override the post_processor specified in the model
                     Default None to disable overriding
-                threads (int, optional): the number of threads to use for training.
+                threads (int, optional): the number of threads to use for prediction.
                     Defaults to -1 to use all
 
         Returns:
             Y_pred (csr_matrix): If a selected output matrix is provided, the prediction for the
                 indicated output is returned. Otherwise, a prediction matrix for some topk
                 scores is returned.
         """
-        if selected_outputs_csr is None:
-            if pred_params is None:
-                Y_pred = self.model.predict(X, pred_params=None, **kwargs)
-            elif isinstance(pred_params, self.PredParams):
-                Y_pred = self.model.predict(X, pred_params=pred_params.hlm_args, **kwargs)
+        if batch_size is None:
+            if selected_outputs_csr is None:
+                if pred_params is None:
+                    Y_pred = self.model.predict(X, pred_params=None, **kwargs)
+                elif isinstance(pred_params, self.PredParams):
+                    Y_pred = self.model.predict(X, pred_params=pred_params.hlm_args, **kwargs)
+                else:
+                    raise TypeError("type(pred_kwargs) is not supported")
             else:
-                raise TypeError("type(pred_kwargs) is not supported")
-        else:
-            if pred_params is None:
-                Y_pred = self.model.predict_on_selected_outputs(
-                    X, selected_outputs_csr, pred_params=None, **kwargs
-                )
-            elif isinstance(pred_params, self.PredParams):
-                Y_pred = self.model.predict_on_selected_outputs(
-                    X, selected_outputs_csr, pred_params=pred_params.hlm_args, **kwargs
+                if pred_params is None:
+                    Y_pred = self.model.predict_on_selected_outputs(
+                        X, selected_outputs_csr, pred_params=None, **kwargs
+                    )
+                elif isinstance(pred_params, self.PredParams):
+                    Y_pred = self.model.predict_on_selected_outputs(
+                        X, selected_outputs_csr, pred_params=pred_params.hlm_args, **kwargs
+                    )
+                else:
+                    raise TypeError("type(pred_kwargs) is not supported")
+        elif isinstance(batch_size, int):
+            Ys = []
+            for i in range(0, X.shape[0], batch_size):
+                Ye = self.predict(
+                    X=X[i : i + batch_size, :],
```

Comment:
`X` is not a keyword argument here.

Reply:
Thanks for the catch. I will fix in the amendment.